common/src/lib.rs

```diff
@@ -64,14 +86,95 @@ pub struct Message {
 }
 
 #[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
-pub struct Program {
+pub enum Program {
+    Active(ActiveProgram),
+    Terminated,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum ProgramError {
+    FailedToGetProgramData,
+    ProgramIsTerminated,
```

Comment:
I'd make them shorter and clearer. Just as an example:
```suggestion
    NoData,
    InitFailed,
```